host/xtest/xtest_6000.c

```diff
@@ -1459,6 +1460,130 @@ static void xtest_tee_test_6015_single(ADBG_Case_t *c, uint32_t storage_id)
 
 DEFINE_TEST_MULTIPLE_STORAGE_IDS(xtest_tee_test_6015)
 
+
+struct test_6016_thread_arg {
+	ADBG_Case_t *case_t;
+	uint32_t storage_id;
+	uint8_t file_name[8];
+	TEEC_Session session;
+};
+
+static void *test_6016_thread(void *arg)
+{
+	struct test_6016_thread_arg *a = arg;
+	TEEC_Session sess = a->session;
+	uint32_t obj;
+	uint8_t out[10] = { 0 };
+	uint32_t count;
+
+	/* create */
+	if (!ADBG_EXPECT_TEEC_SUCCESS(a->case_t,
+		fs_create(&sess, a->file_name, sizeof(a->file_name),
+			  TEE_DATA_FLAG_ACCESS_WRITE, 0, data_01,
+			  sizeof(data_01), &obj, a->storage_id)))
+		goto exit;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(a->case_t, fs_close(&sess, obj)))
+		goto exit;
+
+	/* write new data */
+	if (!ADBG_EXPECT_TEEC_SUCCESS(a->case_t,
+		fs_open(&sess, a->file_name, sizeof(a->file_name),
+			TEE_DATA_FLAG_ACCESS_WRITE, &obj, a->storage_id)))
+		goto exit;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(a->case_t,
+		fs_write(&sess, obj, data_00, sizeof(data_00))))
+		goto exit;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(a->case_t, fs_close(&sess, obj)))
+		goto exit;
+
+	/* verify */
+	if (!ADBG_EXPECT_TEEC_SUCCESS(a->case_t,
+		fs_open(&sess, a->file_name, sizeof(a->file_name),
+			TEE_DATA_FLAG_ACCESS_READ |
+			TEE_DATA_FLAG_ACCESS_WRITE_META, &obj, a->storage_id)))
+		goto exit;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(a->case_t,
+			fs_read(&sess, obj, out, 10, &count)))
+		goto exit;
+
+	(void)ADBG_EXPECT_BUFFER(a->case_t, data_00, 10, out, count);
+
+	/* clean */
+	if (!ADBG_EXPECT_TEEC_SUCCESS(a->case_t, fs_unlink(&sess, obj)))
+		goto exit;
+
+exit:
+	return NULL;
+}
+
+
+#define NUM_THREADS 4
+static void xtest_tee_test_6016_loop(ADBG_Case_t *c, uint32_t storage_id)
+{
+	size_t num_threads = NUM_THREADS;
+	struct test_6016_thread_arg arg[num_threads];
+	pthread_t thr[num_threads];
+	uint32_t orig;
+	size_t i;
+	size_t n;
+	size_t m;
+
+	memset(arg, 0, sizeof(arg));
+
+	for (m = 0; m < num_threads; m++)
+		if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+			xtest_teec_open_session(&arg[m].session,
+				&storage_ta_uuid, NULL, &orig)))
+			goto out;
```

Comment:
Since thr[] doesn't hold valid thread ids we'll have undefined behavior with `pthread_join()` below.